gtsam/navigation/CombinedImuFactor.h

```diff
@@ -54,6 +48,61 @@ typedef ManifoldPreintegration PreintegrationType;
  *     Robotics: Science and Systems (RSS), 2015.
  */
 
+struct GTSAM_EXPORT PreintegrationCombinedParams : PreintegrationParams {
+  Matrix3 biasAccCovariance;    ///< continuous-time "Covariance" describing accelerometer bias random walk
+  Matrix3 biasOmegaCovariance;  ///< continuous-time "Covariance" describing gyroscope bias random walk
+  Matrix6 biasAccOmegaInt;     ///< covariance of bias used for pre-integration
+
+  /// See two named constructors below for good values of n_gravity in body frame
+PreintegrationCombinedParams(const Vector3& n_gravity) :
+  PreintegrationParams(n_gravity), biasAccCovariance(I_3x3), biasOmegaCovariance(
+    I_3x3), biasAccOmegaInt(I_6x6) {
+  }
+
+  // Default Params for a Z-down navigation frame, such as NED: gravity points along positive Z-axis
+  static boost::shared_ptr<PreintegrationCombinedParams> MakeSharedD(double g = 9.81) {
+    return boost::shared_ptr<PreintegrationCombinedParams>(new PreintegrationCombinedParams(Vector3(0, 0, g)));
+  }
+
+  // Default Params for a Z-up navigation frame, such as ENU: gravity points along negative Z-axis
+  static boost::shared_ptr<PreintegrationCombinedParams> MakeSharedU(double g = 9.81) {
+    return boost::shared_ptr<PreintegrationCombinedParams>(new PreintegrationCombinedParams(Vector3(0, 0, -g)));
+  }
+
+  void setBiasAccCovariance(const Matrix3& cov) { biasAccCovariance=cov; }
+  void setBiasOmegaCovariance(const Matrix3& cov) { biasOmegaCovariance=cov; }
+  void setBiasAccOmegaInt(const Matrix6& cov) { biasAccOmegaInt=cov; }
+  
+  const Matrix3& getBiasAccCovariance() const { return biasAccCovariance; }
+  const Matrix3& getBiasOmegaCovariance() const { return biasOmegaCovariance; }
+  const Matrix6& getBiasAccOmegaInt() const { return biasAccOmegaInt; }
+  
+private:
+  /// Default constructor makes unitialized params struct
+  PreintegrationCombinedParams() {}
+
+  /** Serialization function */
+  friend class boost::serialization::access;
+  template <class ARCHIVE>
+    void serialize(ARCHIVE& ar, const unsigned int /*version*/) {
+    ar& BOOST_SERIALIZATION_BASE_OBJECT_NVP(PreintegratedRotation::Params);
+    ar& BOOST_SERIALIZATION_NVP(biasAccCovariance);
+    ar& BOOST_SERIALIZATION_NVP(biasOmegaCovariance);
+    ar& BOOST_SERIALIZATION_NVP(biasAccOmegaInt);
+  }
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+};
+
+  
+#ifdef GTSAM_TANGENT_PREINTEGRATION
```

Comment:
Is there a reason this moved? Maybe move it back as it seems orthogonal

Reply:
Ok, I moved it back to the top